test/sql-luatest/defaults_test.lua

```diff
@@ -56,3 +56,50 @@ g.test_default_in_show_create_table = function()
         func:drop()
     end)
 end
+
+-- Make sure SQL_EXPR functions works as default value.
+g.test_sql_expr_default_func = function()
+    g.server:exec(function()
+        -- No default.
+        local format = {{'a', 'integer'}, {'b', 'integer', is_nullable = true}}
+        local s = box.schema.space.create('t', {format = format})
+        local _ = s:create_index('ii')
+        s:insert({1})
+        local exp = {{1}}
+        t.assert_equals(s:select(), exp)
+        local func_def = {is_deterministic = true, language = 'SQL_EXPR'}
+
+        -- Function with zero arguments.
+        func_def.body = '123 + 456 * 2'
+        box.schema.func.create('DEF1', func_def)
+        format[2].default_func = 'DEF1'
+        s:format(format)
+        s:insert({2})
+        exp[2] = {2, 1035}
+        t.assert_equals(s:select(), exp)
+
+        -- Function with one argument.
+        func_def.body = '11 * a'
+        box.schema.func.create('DEF2', func_def)
+        format[2].default = 123
+        format[2].default_func = 'DEF2'
+        s:format(format)
+        s:insert({3})
+        exp[3] = {3, 11 * 123}
+        t.assert_equals(s:select(), exp)
+
+        -- Function with more than one argument.
+        func_def.body = 'a * b * c'
+        box.schema.func.create('DEF3', func_def)
+        format[2].default_func = 'DEF3'
+        s:format(format)
+        s:insert({4})
+        exp[4] = {4, 123 * 123 * 123}
```

Comment:
I'd rather document this behaviour (at least as an undefined one). Or feel free to create a ticket to reimplement this semantics.

Reply:
I will create an issue after this PR will be pushed to master.